src/PizzaForm.js

```diff
@@ -0,0 +1,151 @@
+import React, { useState, useEffect } from 'react';
+import { Link } from 'react-router-dom';
+import * as yup from 'yup';
+import axios from 'axios';
+
+function PizzaForm() {
+const [pizzaForm, setPizzaForm] = useState({
+    name: '',
+    size: 'small',
+    pepperoni: false,
+    pineapple: false,
+    anchovies: false,
+    ham: false,
+    sausage: false,
+    instructions: ''
+})
+const [disabled, setDisabled] = useState(true);
+const [order, setOrder] = useState();
+const [serverError, setServerError] = useState([]);
+const [inlineError, setInlineError] = useState({
+    name: ''
+});
+const handleNameChange = e => {
+    e.persist();
+    validate(e);
+    setPizzaForm({...PizzaForm, [e.target.name]: e.target.value})
+}
+const handleChange = e => {
+    e.persist();
+    setPizzaForm({...pizzaForm, [e.target.name]: e.target.type === 'checkbox' ? e.target.checked : e.target.value})
+}
+const handleSubmit = e => {
+    e.preventDefault();
+    axios.post('https://reqres.in/api/users', pizzaForm)
+    .then((res) => {
+        setOrder(res.data);
+        setServerError(null);
+        setPizzaForm({
+            name: '',
+            size: 'small',
+            pepperoni: false,
+            pineapple: false,
+            anchovies: false,
+            ham: false,
+            sausage: false,
+            instructions: ''
+        });
+    })
+    .catch((err) => {
+        setServerError('Uh oh, something is wrong');
+    });
+};
+const formSchema = yup.object().shape({
+    name: yup.string().required('must provide a name').min(2, 'must have more than two characters')
+});
+useEffect(() => {
+    formSchema.isValid(pizzaForm).then((valid) => {
+        setDisabled(!valid);
+    });
+}, [pizzaForm]);
+const validate = e => {
+    yup
+        .reach(formSchema, e.target.name)
+        .validate(
+            e.target.type === 'checkbox' ? e.target.checked : e.target.value
+        )
+        .then((valid) => {
+    setInlineError({...inlineError, [e.target.name]: ''});
+        })
+        .catch((err) => {
+            setInlineError({...inlineError, [e.target.name]: err.errors[0]})
+        })
+}
+return (
+    <form onSubmit={handleSubmit}>
+        {serverError && <p>{serverError}</p>}
+
+        <h1>Pizza Time!</h1>
+        <Link to='/'>
+            <button>Back to Home</button>
+        </Link>
+        <label htmlFor='name'>Name: </label>
```

Comment:
You can also nest the input tag inside the label tag and then you don't need the `htmlFor` attribute, just FYI.